oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/ElasticIndexDefinition.java

```diff
@@ -381,4 +408,238 @@ protected IndexDefinition createInstance(NodeState indexDefnStateToUse) {
+             * The minimum number of terms required for the query to be executed. Default is 2.
+             */
+            public int minTerms;
+            /**
+             * The number of candidates to be returned by the query. Default is 100.
+             */
+            public int numCandidates;
+            /**
+             * The type of the query. Default is "hybrid". Currently not used
+             */
+            public String type; // this can be hybrid or vector
+            /**
+             * The similarity threshold used for the query. Default is 0.5.
+             */
+            public float similarityThreshold;
+            /**
+             * The timeout for the query in milliseconds. Default is 5000.
+             */
+            public long timeout;
+
+            public Query() {}
+
+            /**
+             * Constructs a Query from a given NodeState.
+             *
+             * @param name the name of the query
+             * @param queryNode the NodeState containing the query configuration
+             */
+            public Query(String name, NodeState queryNode) {
+                this.name = name;
+                this.serviceUrl = getOptionalValue(queryNode, "serviceUrl", null);
+                this.model = getOptionalValue(queryNode, "model", "semantic");
+                this.prefix = getOptionalValue(queryNode, "prefix", null);
+                this.minTerms = getOptionalValue(queryNode, "minTerms", 2);
+                this.numCandidates = getOptionalValue(queryNode, "numCandidates", 100);
+                this.type = getOptionalValue(queryNode, "type", "hybrid");
+                this.similarityThreshold = getOptionalValue(queryNode, "similarityThreshold", 0.5f);
+                this.timeout = getOptionalValue(queryNode, "timeout", 5000L);
+            }
+
+            /**
+             * Returns {@code true} if the input string is eligible for this query.
+             * @param input the input string
+             * @return {@code true} if the input string is eligible for this query
+             */
+            public boolean isEligibleForInput(String input) {
+                return prefix == null || input.startsWith(prefix);
+            }
+
+            /**
+             * Rewrites the input string by removing the prefix.
+             *
+             * @param input the input string
+             * @return the rewritten input string
+             */
+            public String rewrite(String input) {
+                return prefix == null ? input : input.substring(prefix.length());
+            }
+
+            /**
+             * Checks if the input string has the minimum number of terms required for this query.
+             *
+             * @param input the input string
+             * @return true if the input string has the minimum number of terms
+             */
+            public boolean hasMinTerms(String input) {
+                return minTerms <= input.split(" ").length;
```

Comment:
Can the input string have whitespace separators other than a single space? Like tabs, newlines or multiple whitespaces? Maybe use `\s+` to break up the string?